app/src/main/cpp/skyline/loader/nso.cpp

```diff
@@ -65,4 +68,46 @@ namespace skyline::loader {
         state.process->memory.InitializeRegions(span<u8>{loadInfo.base, loadInfo.size});
         return loadInfo.entry;
     }
+
+    void NsoLoader::PrintRoContentsInfo(const std::vector<u8> &contents) {
+        const boost::regex moduleRegex(R"([a-z]:[\\/][ -~]{5,}\.nss)", boost::regex::icase);
+        const boost::regex fsSdkRegex("sdk_version: ([0-9.]*)");
+        const boost::regex sdkMwRegex("SDK MW[ -~]*");
+
+        std::string contentsRaw(contents.begin(), contents.end());
+        std::string modulePath;
+        if (memcmp(&contents[0], "\x00\x00\x00\x00", 4) == 0) {
+            i32 length;
+            std::memcpy(&length, &contents[4], sizeof(i32));
+
+            if (length > 0)
+                modulePath = reinterpret_cast<const char *>(&contents[4 + sizeof(i32)]);
+        }
+
+        if (modulePath.empty()) {
+            boost::smatch moduleMatch;
+            if (boost::regex_search(contentsRaw, moduleMatch, moduleRegex))
```

Comment:
Coming back to this, doesn't this work?
```suggestion
            if (boost::regex_search(contents.begin(), contents.end(), moduleMatch, moduleRegex))
```
If it does, you can remove the `contentsRaw` variable.

Reply:
This PR isn't actually needed anymore. I used it a bit locally for REing multimedia, so I'll close it

Reply:
Actually I think it will come in handy for determining NVN version later on, also it's a nice feature to have. If you don't want to work on this anymore that's fine, but I'm going to reopen and merge this.